internal/memory/store.go

```diff
@@ -139,3 +166,28 @@ func (s *store) evaluateType(key string, namespace string, lc map[string]interfa
 	}
 	return cfg, rp, nil
 }
+
+// wraps the contents in an object that caches its sha256 hash
+// for easy comparison.
+type updateRequest struct {
+	contents    *backendv1beta1.GetRepositoryContentsResponse
+	contentHash *string
+}
+
+func (ur *updateRequest) hash() error {
```

Comment:
```suggestion
func (ur *updateRequest) calculateContentHash() error {
```
it wasn't immediately clear to me that `hash` is a verb here :)